src/encryption/update.go

```diff
@@ -0,0 +1,50 @@
+package encryption
+
+import (
+	"fmt"
+
+	"github.com/iancoleman/orderedmap"
+)
+
+func updateContentElement(element interface{}, keyPath path, value interface{}) interface{} {
+
+	switch currentElement := element.(type) {
+	case *orderedmap.OrderedMap:
+		childStep := keyPath[0].String()
+		childElement, ok := currentElement.Get(childStep)
+		if !ok {
+			panic(fmt.Errorf("orderedMap \"%v\" missing key %v", element, childStep))
+		}
+		newChildElement := updateContentElement(childElement, keyPath[1:], value)
+		currentElement.Set(childStep, newChildElement)
+		return currentElement
+	case orderedmap.OrderedMap:
+		childStep := keyPath[0].String()
+		childElement, ok := currentElement.Get(childStep)
+		if !ok {
+			panic(fmt.Errorf("orderedMap \"%v\" missing key %v", element, childStep))
+		}
+		newChildElement := updateContentElement(childElement, keyPath[1:], value)
+		currentElement.Set(childStep, newChildElement)
+		return currentElement
+	case []interface{}:
+		childStep := keyPath[0].(sliceStep)
+		if int(childStep) >= len(currentElement) {
+			panic(fmt.Errorf("slice \"%v\" index %v out of bouds", element, childStep))
+		}
+		childElement := currentElement[childStep]
+		newChildElement := updateContentElement(childElement, keyPath[1:], value)
+		currentElement[childStep] = newChildElement
+		return currentElement
+	// scalar value
+	default:
+		if len(keyPath) > 0 {
+			panic(fmt.Errorf("scalar value update with non-empty path found %v ", keyPath))
+		}
+		return value
+	}
+}
+
+func UpdateContent(content *orderedmap.OrderedMap, keyPath path, value interface{}) *orderedmap.OrderedMap {
```

Comment:
Mozes prosim pridat nad tuto (alebo druhu) funkciu nejaky komentar co to robi?
Mozno by tomu pomohl aj nejaky lepsi nazov, napr. 
`func SetNestedKey(m *orderedmap.OrderedMap, key path, value interface{}) *orderedmap.OrderedMap {`

Reply:
Inak mohlo by to kludne byt aj v `utils/map.go - SetNestedKey`, aj ked sa to teraz pouziva iba z jedneho balicka, a niektori neodporucaju mat balicky typu `utils` ... do `utils` som daval nejaky pomocny kod, ktory by mohol byt kludne vlastnym balickom a nema priamy suvis s tym co sa deje.

Reply:
ok presuniem to do `utils/map.go`. Ten nazov `SetNestedKey` mi ale uplne nesedi, skor evokuje ze sa nastavuje kluc nez hodnota. Napada ma jednoducho `SetIn`, to mam ale zazite z js/UI kde su presne nato podobne fukncie `Set(key, value)` a `SetIn(keysArray[], value)`.  Pridam aj komentar

Reply:
ok ;)

Reply:
mi doslo ze tym presunutim budem musiet presunut aj cely encryption/path.go subor:
https://github.com/keboola/keboola-as-code/blob/kacurez-encrypt-update-content-SRE-2479/src/encryption/path.go
V utils uz taky subor path.go je tak by som ho tam prekopiroval 
https://github.com/keboola/keboola-as-code/blob/kacurez-encrypt-update-content-SRE-2479/src/utils/path.go
a vystavil jeho types public pretoze sa pouzivaju aj v encryption package:
https://github.com/keboola/keboola-as-code/blob/kacurez-encrypt-update-content-SRE-2479/src/encryption/encryption.go

Reply:
@kacurez v utils su 2 path,
- `PathTemplate` ... tj. sablona pre nejaku cestu na disku
- `PathState` - stav nejakej cesty na disku

Ty by si to mohol mazvaz `KeyPath` .... a aspon by si mohol v kode pouzivat premenne s nazvom `path` :)

Reply:
teda nemam nic proti, len treba dat tomu nejaky nazov, iba `Path` (ked to uz nebude v `encrypt`) asi nebude dost jednoznacne.

Reply:
*** omyl, to `PathState` je v `state` balicku, sorry (ale to vlastne nic nemeni).

Reply:
ok, premenujem to na `KeyPath`